unidist/core/backends/mpi/core/controller/common.py

```diff
@@ -255,3 +256,76 @@ def push_data(dest_rank, value):
             _push_data_owner(dest_rank, value)
         else:
             raise ValueError("Unknown DataID!")
+
+
+def push_data_owners(dest_rank, data_ids):
+    """
+    Parse and send data owners for all values to destination rank.
+
+    Send all ID associated location to the target rank.
+
+    Parameters
+    ----------
+    dest_rank : int
+        Rank where task data is needed.
+    data_ids : list
+        List of all data_ids required by the worker
+    """
+    for data_id in data_ids:
+        if object_store.contains_data_owner(data_id):
+            _push_data_owner(dest_rank, data_id)
+        else:
+            raise ValueError("Data owner not known")
+
+
+def choose_destination_rank(data_ids):
+    """
+    Choose destination rank considering which worker has the maximum share of data.
+
+    Get the data shares in each worker process and choose rank with the most data share.
+    If data_ids are empty choses the rank using the round robin approach.
+
+    Parameters
+    ----------
+    data_ids : list
+        List of all data_ids required by the worker
+
+    Returns
+    -------
+    int
+        A rank number.
+    """
+    if not data_ids:
```

Comment:
When is data_ids an empty list?

Reply:
This would for the case where there are no arguments for the remote function. So the data_ids would be empty and none of workers will have data shares so submitting them using round robin.

eg.

```
@unidist.remotedef foo():
    return 1 
o = foo.remote()
```

Reply:
Put a comment describing this into the code.